src/_guides/openapi/code-first.md

```diff
@@ -0,0 +1,177 @@
+
+There are not as many tools that work this way, but this is likely to be a trend that continues as OpenAPI becomes the dominant API description format.
+
+- **Go** - [Huma](https://huma.rocks/) ([Guide](/guides/bump-sh-tutorials/huma/))
+- **PHP** - [API Platform](https://api-platform.com/) ([Guide](/guides/bump-sh-tutorials/api-platform/))
+- **Python** - [FastAPI](https://fastapi.tiangolo.com/) ([Guide](/guides/bump-sh-tutorials/fastapi/))
+
+Just like annotations you can usually run a command to extract the OpenAPI document, or you can run the web server and pull it down over HTTP.
+
+```
+$ go run .
+
+$ bump deploy http://127.0.0.1:8888/openapi.yaml \
+  --doc my-documentation-name \
+  --token my-documentation-token
+```
+
+## Traffic Sniffing
+
+If there's no annotations approach, and you have an existing codebase which cannot be rebuilt with one of these OpenAPI-aware application frameworks, there is another powerful option: sniffing web traffic. 
+
+There's [a whole category of tools](https://openapi.tools/#learning) popping up, which refer to this functionality as "Recording" or "Learning". 
+
+Basically you run an instance of your API somewhere (could be local, test, staging, or even production) and put as much web traffic through it as possible. It will then learn how all the requests and responses look, and produce the best composite OpenAPI that it possibly can.
+
+This is mostly useful as a one-off, a way to produce a bunch of OpenAPI that you then manage and maintain yourself, because running this forever is not a sensible workflow and will generally only get things 90% right. You need to do the work to get to 100%, but at least you didn't have to do all of the work of making that initial OpenAPI for an API that already exists. That's going to be arduous, boring, and likely rife with human error.
+
+## Move to API Design-first
+
+With so many of the annotations approaches being outdated, and people usually unable to rebuild an entire codebase to use a framework that happens to emit OpenAPI, a lot of people have given up on the whole code-first approach. 
+
+This is not just an opinion. Searching around the Go community for code-first tooling, most of the "How do I do Code-first in Go" search results show people talking about how they moved to API Design-first and massively prefer the approach.
+
+The main idea is that instead of writing loads of code and sprinkling in some annotations later to create docs, you create the OpenAPI before writing any code at all. This is usually in the form of JSON/YAML, but that does not need to be written by hand. There are lots of visual editors to help you build this all up through buttons and forms, with an increasing amount of intelligence to create things. If you're using VS Code then Copilot is actually incredibly good.  
```

Comment:
```suggestion
The main idea is that instead of writing loads of code and sprinkling in some annotations later to create docs, you create the OpenAPI document before writing any code at all. This is usually in the form of JSON/YAML, but that does not need to be written by hand. There are lots of visual editors to help you build this all up through buttons and forms, with an increasing amount of intelligence to create things. If you're using VS Code then Copilot is actually incredibly good for instance.
```